test/e2e/e2e_suite_test.go

```diff
@@ -47,6 +47,8 @@ var (
 	// clusterProvider manages provisioning of the cluster to be used for the e2e tests.
 	// Please note that provisioning will be skipped if e2e.use-existing-cluster is provided.
 	clusterProvider bootstrap.ClusterProvider
+
+	kubeconfigPath string
```

Comment:
This is a bit dangerous to make global. We update it if/when creating a kind cluster and this then affects the variable across all tests. For now I don't think it is an issue since we use the clusterproxy instead of the kubeconfig in most tests. However, I don't want to rely on luck here. There could be tests in the future when we use the kubeconfig. So please put it back as a local variable and use a separate local variable for the BMO upgrade test. Then they won't affect each other.